kratos/linear_solvers/fallback_linear_solver.h

```diff
@@ -0,0 +1,895 @@
+    {
+        // Verify that linear solvers are not defined in the parameters
+        KRATOS_ERROR_IF(mParameters.Has("solvers")) << "The solvers are already defined in the input parameters" << std::endl;
+
+        // Set the default parameters
+        mParameters.ValidateAndAssignDefaults(GetDefaultParameters());
+
+        // Fill the parameters with the solvers
+        for (auto& p_solver : mSolvers) {
+            FillParametersFromSolver(p_solver);
+        }
+
+        // Set some member variables from the parameters
+        CommonSettingsFromParameters();
+    }
+
+    /// Copy constructor.
+    FallbackLinearSolver(const FallbackLinearSolver& rOther) = delete;
+
+    /// Destructor.
+    ~FallbackLinearSolver() override = default;
+
+    ///@}
+    ///@name Operators
+    ///@{
+
+    /// Assignment operator.
+    FallbackLinearSolver& operator=(const FallbackLinearSolver& rOther) = delete;
+
+    ///@}
+    ///@name Operations
+    ///@{
+
+    /**
+     * @brief This function is designed to be called as few times as possible.
+     * @details It creates the data structures that only depend on the connectivity of the matrix (and not on its coefficients) so that the memory can be allocated once and expensive operations can be done only when strictly  needed
+     * @param rA. System matrix
+     * @param rX. Solution vector. it's also the initial guess for iterative linear solvers.
+     * @param rB. Right hand side vector.
+     */
+    void Initialize(
+        SparseMatrixType& rA,
+        VectorType& rX,
+        VectorType& rB
+        ) override
+    {
+        // TODO: Decide if call all solvers or just the current one
+        // NOTE: We assume that initializes internal data of solvers, not system of equations
+        for (auto& p_solver : mSolvers) {
+            p_solver->Initialize(rA, rX, rB);
+        }
+        // GetCurrentSolver()->Initialize(rA, rX, rB);
+    }
+
+    /**
+     * @brief This function is designed to be called every time the coefficients change in the system that is, normally at the beginning of each solve.
+     * @details For example if we are implementing a direct solver, this is the place to do the factorization
+     * so that then the backward substitution can be performed effectively more than once
+     * @param rA. System matrix
+     * @param rX. Solution vector. it's also the initial guess for iterative linear solvers.
+     * @param rB. Right hand side vector.
+     */
+    void InitializeSolutionStep(
+        SparseMatrixType& rA,
+        VectorType& rX,
+        VectorType& rB
+        ) override
+    {
+        // TODO: Decide if call all solvers or just the current one
+        // NOTE: We assume it may modify the system of equations, therefore we call just in the current solver
+        // for (auto& p_solver : mSolvers) {
+        //     p_solver->InitializeSolutionStep(rA, rX, rB);
+        // }
+        GetCurrentSolver()->InitializeSolutionStep(rA, rX, rB);
```

Comment:
this should be consistent!

Reply:
But if InitializeSolutionStep modifies the LHS and RHS, then it will break the solution. It is not easy to manage this.

Reply:
This is called later when switching the solver